### PR TITLE
Health system

### PR DESCRIPTION
**Pull request by keys-i**
# Description
This PR improves the **Health System tests** and fixes existing issues in `CombatStatsComponent`. The key changes are below:

1. **Refactored tests** into nested classes aligned with the Health System test plan for clearer structure and coverage.
2. **Added new unit tests** for `hit(int)`, `hit(CombatStatsComponent)`, `addHealth`, death-state handling, and event routing.
3. **Fixed bugs** in `setHealth` (now clamps to `0` instead of overwriting `maxHealth`) and `getMaxHealth` (returns `this.maxHealth` correctly).
4. **Exposed and documented edge cases**: inconsistent revive semantics, overheal behaviour, and duplicate event emission.

All existing tests pass, and the new suite provides stronger regression coverage of health, attack, death, and event logic.

Fixes / Closes # (#4)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

- [ ] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

- [x] This change requires a documentation update

# How Has This Been Tested?

All existing `CombatStatsComponentTest` cases continue to pass after the changes. The suite was refactored into nested classes based on the test plan, and new tests were added.

* [x] **Initialisation** – verifies `health`, `baseAttack`, `maxHealth`, and `isDead` on construction, including zero and negative inputs (captures known ctor bug).
* [x] **setHealth** – clamps to zero, updates to positive values, fires `"updateHealth"` once per change, and marks death state correctly.
* [x] **addHealth** – delegates to `setHealth`, supports healing and damage, clamps on overkill, handles zero/no-op consistently.
* [x] **Death behaviour** – confirms dead entities ignore further damage and remain `isDead()`.
* [x] **Direct damage (hit(int))** – reduces health appropriately, clamps to zero, ignores non-positive values, accumulates across multiple hits.
* [x] **Attacker damage (hit(attacker))** – applies `baseAttack` via `applyDamage()`, ignores null attacker, rejects negative `baseAttack`, respects updated values at runtime.
* [x] **setMaxHealth defect** – tests added to expose incorrect assignment of `health` to `maxHealth`; it's fixed.

**Reproduction steps:**

1. Build with JDK 17 using the provided Gradle wrapper.
2. Run:

   ```bash
   ./gradlew test
   ```
3. Open `source/core/build/reports/tests/test/index.html` to verify all tests pass.

# Checklist:

- [x] My code follows the style guidelines of this project

- [x] I have performed a self-review of my code

- [x] I have commented my code, particularly in hard-to-understand areas

- [x] I have made corresponding changes to the documentation

- [x] My changes generate no new warnings

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] New and existing unit tests pass locally with my changes

- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by jholmesuq — APPROVED** [review]

File: source/core/src/main/com/csse3200/game/components/CombatStatsComponent.java
[the remark this answers is not in the document]
Comment (keys-i) [reply source/core/src/main/com/csse3200/game/components/CombatStatsComponent.java:1]: I have updated it to be the upper bound for Health

**Review by jholmesuq — APPROVED** [review]

File: source/core/src/main/com/csse3200/game/components/CombatStatsComponent.java [hunk 1]
```diff
@@ -42,23 +54,17 @@ public Boolean isDead() {
    *
```
Comment (cailindelgado) [inline source/core/src/main/com/csse3200/game/components/CombatStatsComponent.java:54]: Java Doc required

**Review by cailindelgado — CHANGES_REQUESTED** [review]
Overall looks good, checkout the comment I left for missing docstrings.

**Review by cailindelgado — CHANGES_REQUESTED** [review]
Still need to fix the docstrings for getHealth(), and the formatting for it
